### PR TITLE
Implement Cross-Validation

### LINKED ISSUE #103: Implement Cross-Validation
- [ ] train_test_split: Folds produzieren
- [ ] model_runner: Cross-Validation implementieren
- [ ] Runs nochmal durchführen um Model Performance Evaluation zu erhalten

### PR DESCRIPTION
**Pull request by gabrieltorresgamez**
#103 

@buesst1 sorry dassi wieder dir gib zum reviewe, aber da du ja das mit de modellbranches machsch würs sich am meiste lohne wennis dir gib

### CODE REVIEW

**Comment by ghost** [comment]
# RUN: HistGradientBoostingClassifier, 2023-06-08 22-48-25

## Hyperparameters

| Path        | Param                            | main   | workspace        |
|-------------|----------------------------------|--------|------------------|
| params.yaml | cross_validation_split.cv_splits | -      | 10               |
| params.yaml | cross_validation_split.seed      | -      | 9458             |
| params.yaml | cross_validation_split.type      | -      | cross_validation |
| params.yaml | dvclive.seed                     | 41     | 42               |

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix

### CV Metrics

CV Metrics

File: src/train_test_split.py [hunk 1]
```diff
@@ -66,7 +74,34 @@
...
+    elif split_type == "cross_validation":
+        splits = params["cv_splits"]
+
+        kfolds = KFold(n_splits=splits, shuffle=True, random_state=random_seed).split(
+            df
+        )
+
+        # create a train-test split for each split
+        for i in range(splits):
+            train, test = next(kfolds)
```
Comment (buesst1) [inline src/train_test_split.py]: Wiso mit next()?

Chönnt me nid einfach folgende code verwende?

for i, (train, test) in enumerate(kfolds):
   ...

**Review by buesst1 — CHANGES_REQUESTED** [review]
Gud.

-> siehe comment

File: src/train_test_split.py (hunk 1 above)
Comment (gabrieltorresgamez) [reply src/train_test_split.py]: sure man, beides gaht, hans jetzt mit enumerate ersetzt

**Review by buesst1 — APPROVED** [review]
purrfect

**Comment by ghost** [comment]
# RUN: HistGradientBoostingClassifier, 2023-06-09 09-42-57

## Hyperparameters

| Path   | Param   | main   | workspace   |
|--------|---------|--------|-------------|

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix

### CV Metrics

CV Metrics
